SIMS/SIMS/Inventory.cs

```diff
@@ -0,0 +1,122 @@
+namespace SIMS;
+
+public class Inventory
+{
+    // Products property
+    private List<Product> Products { get; set; } = [];
+
+    // AddProduct method
+    public bool AddProduct(Product? product)
+    {
+        //check if the product is null
+        if (product is null)
+        {
+            return false;
+        }
+
+        //check if the product already exists
+        if (Products.Any(p => p.Name == product.Name))
+        {
+            return false;
+        }
+
+        //add the product to the list
+        Products.Add(product);
+        return true;
+    } // End of AddProduct method
+
+    //ViewAllProducts method
+    public void ViewAllProducts()
+    {
+        //check if there are no products
+        if (Products.Count == 0)
+        {
+            Console.WriteLine("No products found");
+        }
+
+        //print all the products
+        foreach (var product in Products)
+        {
+            Console.WriteLine(product);
+        }
+    } // End of ViewAllProducts method
+
+    //UpdateProduct method
+    public bool UpdateProduct(string currentProductName, string? newProductName,
+        decimal? newProductPrice, int? newProductQuantity)
+    {
+        //find the product
+        var product = FindProduct(currentProductName);
+
+        //check if the product is null
+        if (product is null)
+        {
+            return false;
+        }
+
+        if (!string.IsNullOrEmpty(newProductName))
+        {
+            product.Name = newProductName;
+        }
+
+        if (newProductPrice.HasValue)
+        {
+            product.Price = newProductPrice.Value;
+        }
+
+        if (newProductQuantity.HasValue)
+        {
+            product.Quantity = newProductQuantity.Value;
+        }
+
+        return true;
+    } // End of UpdateProduct method
+
+    //FindProduct method
+    private Product? FindProduct(string name)
+    {
+        //find the product
+        return Products.Find(p => p.Name!.Equals(name, StringComparison.OrdinalIgnoreCase));
+    } // End of FindProduct method
+
+    //DeleteProduct method
+    public bool DeleteProduct(string name)
+    {
+        //find the product
+        var product = FindProduct(name);
+
+        //check if the product is null
+        if (product is null)
+        {
+            return false;
+        }
+
+        //remove the product
+        Products.Remove(product);
+        return true;
+    } // End of DeleteProduct method
+
+    //SearchProduct method
+    public void SearchProduct(string name)
```

Comment:
what is the responsibility of this function? according to your code it's for printing a specific product, but the method's name does not imply that. What if another developer works with you in the same project, and he wants to use your function, he will expect this function to return him a `Product` instance since he's searching about one. Please reconsider your implementation.